src/promptflow/promptflow/_sdk/operations/_run_operations.py

```diff
@@ -228,6 +228,31 @@ def update(
         ORMRun.get(name).update(display_name=display_name, description=description, tags=tags, **kwargs)
         return self.get(name)
 
+    @monitor_operation(activity_name="pf.runs.delete", activity_type=ActivityType.PUBLICAPI)
+    def delete(
+        self,
+        runs: Union[str, Run, List[str], List[Run]],
+        **kwargs,
+    ) -> None:
+        """Delete run permanently.
+
+        :param names: run names to delete
+        :param kwargs: other fields to update, fields not supported will be directly dropped.
+        :return: None
+        """
+        if not isinstance(runs, list):
+            runs = [runs]
+        valid_runs = []
+        for run in runs:
+            run = Run._validate_and_return_run_name(run)
+            valid_runs.append((run, LocalStorageOperations(run=self.get(run))))
+        for run, valid_run in valid_runs:
+            valid_run.delete()
```

Comment:
The internal deletion function may throw an exception? Do we need to catch it?
![image](https://github.com/microsoft/promptflow/assets/23182548/303c7c30-5d12-4551-95e5-3f0bb12bb8ce)

Reply:
Yes. let make a small discussion offline.